lib/src/main/kotlin/WeatherSensorService/Internal/MockWeatherSensorReader.kt

```diff
@@ -0,0 +1,69 @@
+ //
+ // SpinDance
+ // Weather Sensor Service
+ //
+ // Copyright © 2022 SpinDance. All rights reserved.
+ //
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import java.util.Timer
+import kotlin.concurrent.*
+import kotlin.random.Random
+import java.time.LocalDateTime
+
+private class Constants {
+    companion object {
+        var READER_INTERVAL: UInt = 1U
+        val TEMPERATURE_RANGE : ClosedRange<Double> = -40.0..40.0
+        val HUMIDITY_RANGE : ClosedRange<Double> = 0.0..100.0
+        val PRESSURE_RANGE : ClosedRange<Double> = 95.0..105.0
+    }
+}
+
+class MockWeatherSensorReader: WeatherSensorReaderType {
+     
+    private var timer: Timer? = null
+
+    override var readerInterval : UInt = Constants.READER_INTERVAL
+        
+    
+    private val flow: MutableSharedFlow<WeatherSensorReading> = MutableSharedFlow<WeatherSensorReading>(replay = 1)
+    override var sensorReadingsPublisher : SharedFlow<WeatherSensorReading> = flow.asSharedFlow()
+    
+    override fun set(readingInterval: UInt) {
+        if (readingInterval != readerInterval && timer != null){
+            startSensorReadings()
+        }
+        if (readingInterval >= 1U && readingInterval <= UInt.MAX_VALUE) { 
+            readerInterval = readingInterval 
+        }
```

Comment:
Seems like we should not start readings if the readingInterval is out of range. Does the caller have a way of knowing that their input is out of range? Something like:
```
if (<interval is not changing> || <interval is out of range> { 
    return
}

this.readingInterval = readingInterval

if (timer != NULL) {
    startReading()
}
```